.github/workflows/build.yml

```diff
@@ -28,6 +28,76 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  prebuild-legion:
+    name: Prebuild Legion with CMake
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        shell: bash -l {0} # required to use an activated conda environment
+    strategy:
+      matrix:
+        gpu_backend: ["cuda", "hip_rocm"]
+      fail-fast: false
+    steps:
+      - name: Checkout Git Repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Free additional space on runner
+        run: .github/workflows/helpers/free_space_on_runner.sh
+
+      - name: Install CUDA
+        uses: Jimver/cuda-toolkit@v0.2.11
+        id: cuda-toolkit
+        with:
+          cuda: "11.8.0"
+          # Disable caching of the CUDA binaries, since it does not give us any significant performance improvement
+          use-github-cache: "false"
+
+      - name: Install system dependencies
+        run: FF_GPU_BACKEND=${{ matrix.gpu_backend }} .github/workflows/helpers/install_dependencies.sh
+
+      - name: Install conda and FlexFlow dependencies
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: flexflow
+          environment-file: conda/environment.yml
+          auto-activate-base: false
+        
+      - name: Build Legion
+        env: 
+          FF_GPU_BACKEND: ${{ matrix.gpu_backend }}
+        run: |
+          export PATH=/opt/conda/bin:$PATH
+          export CUDNN_DIR=/usr/local/cuda
+          export CUDA_DIR=/usr/local/cuda
+          export FF_CUDA_ARCH=all
+          export BUILD_LEGION_ONLY=ON
+
+          # create and export the installation path
+          export INSTALL_DIR="export/legion"
+          mkdir -p export/legion
+
+          cores_available=$(nproc --all)
+          n_build_cores=$(( cores_available -1 ))
+
+          mkdir build
+          cd build
+          ../config/config.linux
+          make -j $n_build_cores
+          ../config/config.linux
+          make install
+           
+          # prepare library files extract LEGION tarball file 
+          export LEGION_TARBALL="legion_ubuntu-20.04_${{ matrix.gpu_backend }}.tar.gz"
+          echo "Creating archive $LEGION_TARBALL"
+          cd export
+          touch "$LEGION_TARBALL"
+          tar --exclude="$LEGION_TARBALL" -zcvf $LEGION_TARBALL .
```

Comment:
why do we need the `--exclude="$LEGION_TARBALL"` argument?

I think the second argument of the tar command should be the folder name (`legion`) not `.`

Reply:
If we don't use '--exclude="$LEGION_TARBALL"', it will report the error that 'file changed as we read it'.